pkg/statsd/handler_cloud.go

```diff
@@ -166,15 +225,28 @@ func (ch *CloudHandler) doRefresh(ctx context.Context, t time.Time) {
 	var toDelete []gostatsd.IP
 	now := t.UnixNano()
 	idleNano := ch.cacheOpts.CacheEvictAfterIdlePeriod.Nanoseconds()
+
+	positiveDeleted := uint64(0)
+	negativeDeleted := uint64(0)
+
 	for ip, holder := range ch.cache {
 		if now-holder.lastAccess() > idleNano {
 			// Entry was not used recently, remove it.
 			toDelete = append(toDelete, ip)
+			if holder.instance == nil {
+				negativeDeleted++
+			} else {
+				positiveDeleted++
+			}
 		} else if t.After(holder.expires) {
 			// Entry needs a refresh.
 			ch.toLookupIPs = append(ch.toLookupIPs, ip)
 		}
 	}
+
+	atomic.AddUint64(&ch.statsCachePositive, ^(positiveDeleted - 1))
+	atomic.AddUint64(&ch.statsCacheNegative, ^(negativeDeleted - 1))
```

Comment:
How does this work?

Reply:
Why: https://golang.org/pkg/sync/atomic/#AddUint64
How: https://en.wikipedia.org/wiki/Two's_complement

I considered throwing a function the gostatsd package or something, but it seemed kinda meh.

Reply:
I think I'd like this refactored out into another function - I doubt I'll remember this if I came across it in a few months

Reply:
But it's idiomatic!

Seriously though, I've removed most of the atomics in a refactor and we don't need to decrement anymore.